spec/graphql/resolvers/groups_spec.rb

```diff
@@ -0,0 +1,136 @@
+require 'rails_helper'
+
+RSpec.describe Resolvers::Groups, type: :request do
+  let!(:owner) { create :user, :owner }
+  let!(:public_group_1) { create :group, owner: owner }
+  let!(:public_group_2) { create :group, owner: owner }
+  let!(:private_group) { create :group, owner: owner, private: true }
+  let!(:archive_group) { create :group, owner: owner, archived: true }
+  
+  let!(:attendee) { create :user, :attendee }
+  let!(:attenable_1) { create :attendee, attendee: attendee, resource: public_group_1 }
+  let!(:attenable_2) { create :attendee, attendee: attendee, resource: private_group }
+  let!(:token) { generate_jwt_test_token(attendee) }
+
+  let!(:query) do
+    <<-GQL
+      query($id: String!, $range: String!){
+        groups (id: $id, range: $range){
+          id,
+          name,
+          description,
+          slots,
+          country,
+          region,
+          city,
+          streetAddress,
+          postCode,
+          private,
+          attend
+        }
+      }
+    GQL
+  end
+
+  describe 'range is empty' do
+    context 'without specific id' do
+      let!(:variables) do
+        {
+          id: '',
+          range: ''
+        }
+      end
+
+      it 'fetches all active public groups' do
+        post '/graphql', params: { query: query, variables: variables }, headers: { 'Authorization' => "Bearer #{token}" }
```

Comment:
I don't understand what you mean for variables part.
can you explain more?

Reply:
@VinceYoon sure, my idea is making `execute_and_parse_graphql_response` method execute `FocusSchema.execute()` without variables if the variables not passed. but I tried and didn't work because `query` is not a hash format:
```
def execute_and_parse_graphql_response(query:, variables: nil, current_user: nil)
      params = {query, context: { current_user: current_user }}
      params.merge(variables) if variables.present?
      result = FocusSchema.execute(params)
```
maybe only way can do is using `**kwargs` for `variables`. I'm still considering I think it's better to use same method for every cases. how do you think?